src/unitquaternion.jl

```diff
@@ -197,77 +211,70 @@ Rotations.params(p) == @SVector [1.0, 2.0, 3.0]  # true
 """
 @inline params(q::UnitQuaternion) = SVector{4}(q.w, q.x, q.y, q.z)
 
+# TODO: this will be removed, because Quaternion is not a subtype of Rotation
+@inline params(q::Quaternion) = SVector{4}(q.s, q.v1, q.v2, q.v3)
+
 # ~~~~~~~~~~~~~~~ Initializers ~~~~~~~~~~~~~~~ #
 function Random.rand(rng::AbstractRNG, ::Random.SamplerType{<:UnitQuaternion{T}}) where T
-    normalize(UnitQuaternion{T}(randn(rng,T), randn(rng,T), randn(rng,T), randn(rng,T)))
+    _normalize(UnitQuaternion{T}(randn(rng,T), randn(rng,T), randn(rng,T), randn(rng,T)))
 end
 @inline Base.one(::Type{Q}) where Q <: UnitQuaternion = Q(1.0, 0.0, 0.0, 0.0)
 
 
 # ~~~~~~~~~~~~~~~ Math Operations ~~~~~~~~~~~~~~~ #
 
 # Inverses
-conj(q::Q) where Q <: UnitQuaternion = Q(q.w, -q.x, -q.y, -q.z, false)
-inv(q::UnitQuaternion) = conj(q)
-(-)(q::Q) where Q <: UnitQuaternion = Q(-q.w, -q.x, -q.y, -q.z, false)
+inv(q::Q) where Q <: UnitQuaternion = Q(q.w, -q.x, -q.y, -q.z, false)
 
 # Norms
-LinearAlgebra.norm(q::UnitQuaternion) = sqrt(q.w^2 + q.x^2 + q.y^2 + q.z^2)
 vecnorm(q::UnitQuaternion) = sqrt(q.x^2 + q.y^2 + q.z^2)
```

Comment:
The `vecnorm` must be defined in terms of the matrix. Which is orthogonal, so that's easy. Should we just define this instead?

```julia
vecnorm(::Rotation{N, T}) where {N, T} = convert(N, T)
```

Reply:
The `vecnorm` function was to calculate the norm of the imaginary part of a quaternion, and it was used only in `_log_as_quat`. So, I've removed the function. (https://github.com/JuliaGeometry/Rotations.jl/pull/175/commits/09e03135986c7e3dcacf7425f2d0e19b94867ba1)